sonic_y_cable/y_cable_base.py

```diff
@@ -0,0 +1,1130 @@
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+            target:
+                One of the following predefined constants, the actual target to check the link on:
+                     TARGET_NIC -> NIC,
+                     TARGET_TOR_A -> TORA,
+                     TARGET_TOR_B -> TORB
+
+        Returns:
+            a boolean, True if the link is active
+                     , False if the link is not active
+        """
+
+        raise NotImplementedError
+
+    def get_eye_heights(self, target):
+        """
+        This API returns the EYE height value for a specfic port.
+        The target could be local side, TOR_A, TOR_B, NIC etc.
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+            target:
+                 One of the following predefined constants, the target on which to get the eye:
+                     EYE_PRBS_LOOPBACK_TARGET_LOCAL -> local side,
+                     EYE_PRBS_LOOPBACK_TARGET_TOR_A -> TOR A
+                     EYE_PRBS_LOOPBACK_TARGET_TOR_B -> TOR B
+                     EYE_PRBS_LOOPBACK_TARGET_NIC -> NIC
+        Returns:
+            a list, with EYE values of lane 0 lane 1 lane 2 lane 3 with corresponding index
+        """
+
+        raise NotImplementedError
+
+
+    def get_vendor(self):
+        """
+        This API returns the vendor name of the Y cable for a specfic port.
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+
+        Returns:
+            a string, with vendor name
+        """
+
+        raise NotImplementedError
+
+    def get_part_number(self):
+        """
+        This API returns the part number of the Y cable for a specfic port.
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+
+        Returns:
+            a string, with part number
+        """
+
+        raise NotImplementedError
+
+    def get_switch_count_total(self, switch_count_type, clear_on_read=False):
+        """
+        This API returns the total switch count to change the Active TOR which has
+        been done manually/automatic by the user.
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+            switch_count_type:
+                One of the following predefined constants, for getting the count type:
+                    SWITCH_COUNT_MANUAL -> manual switch count
+                    SWITCH_COUNT_AUTO -> automatic switch count
+            clear_on_read:
+                a boolean, True if the count has to be reset after read to zero
+                         , False if the count is not to be reset after read
+            Returns:
+                an integer, the number of times the Y-cable has been switched
+        """
+
+        raise NotImplementedError
+
+    def get_switch_count_tor_a(self, clear_on_read=False):
+        """
```

Comment:
Both get_switch_count_tor_a() and get_switch_count_tor_b() are not supported in current MCU firmware. The cable can only return the total auto/manual switch count by reading the upper page 0x5, byte 141 to 148. If this information is useful for diagnostics, I will update the API document and  MCU FW to support these two APIs from next firmware release.

Reply:
sounds good

Reply:
Originally, we defined a total manual switching count in upper page 0x4, byte 141-144. In order to support these two APIs

- get_switch_count_tor_a()
- get_switch_count_tor_b()

I'd like to change the definition as below

- **upper page 0x4, byte 141-144**: number of manual switching count initiated from **TOR A**
- **upper page 0x4, byte 225-229**: number of manual switching count initiated from **TOR B**

and the get_switch_count_total() will return the sum of TOR A (byte 141-144) and TOR B (byte 225 - 228).